staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/sample_and_watermark_test.go

```diff
@@ -36,6 +35,17 @@ const (
 	numIterations         = 100
 )
 
+var (
+	t0      = time.Now()
+	clk     = testclock.NewFakePassiveClock(t0)
+	buckets = []float64{0, 1}
+	gen     = NewSampleAndWaterMarkHistogramsGenerator(clk, samplingPeriod,
+		&compbasemetrics.HistogramOpts{Name: samplesHistName, Buckets: buckets},
+		&compbasemetrics.HistogramOpts{Name: "marks", Buckets: buckets},
+		[]string{})
+	dt time.Duration
+)
```

Comment:
with a frame-local registry, these can remain frame-local too.

Reply:
In fact, it would be much tidier to keep these local to the test func that uses them.

Reply:
I remember I tried before. It didn't work.

https://gist.github.com/CatherineF-dev/ea06940aa9101f428685560f83c67cc2

Reply:
That gist pointer only shows me some code, not testing results.
I tried testing a simple fix myself, and that revealed a bug in the test.
See #105886 for both a bug fix to the test and support for repeating the test.